Improve demo extension documentation

aspiers opened the pull request:
Fixes #41.

aspiers committed:
Update demo readme.md to match latest code

Fixes #41.

aspiers committed:
Add a hint to demo/readme.md about Developer mode.

fregante reviewed (changes requested):
The indentation breaks the formatting. Please remove the extra tabs/spaces

![image](https://github.com/fregante/webext-permission-toggle/assets/1402241/86324a00-abbc-4ed3-a2e8-2ba009897612)

fregante commented:
As for these changes, I appreciated the effort, but I prefer the previous version. This is a playground for the module, I don't need to document how to build and run the extension. 

I do like the extra commands to run the mv2/3 extension in the specific browser.

aspiers commented:
Hmm, that's weird - could have sworn that indentation has always worked for me before.  Anyway I've fixed it now.

I also removed the sentence about what `npm run watch` does.  Should I remove anything else?

My logic for adding the hints about how the demo works and how to debug it was that it took me some time to figure these out, so I thought that it would also help others who want to contribute to this extension.
